src/net_processing.cpp

```diff
@@ -938,11 +938,12 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const CValidationSta
     // 3. This is currently the best block we're aware of. We haven't updated
     //    the tip yet so we have no way to check this directly here. Instead we
     //    just check that there are currently no other blocks in flight.
-    else if (state.IsValid() &&
-             !IsInitialBlockDownload() &&
-             mapBlocksInFlight.count(hash) == mapBlocksInFlight.size()) {
-        if (it != mapBlockSource.end()) {
-            MaybeSetPeerAsAnnouncingHeaderAndIDs(it->second.first, connman);
+    else if (state.IsValid()) {
+        initialBlockDownloadCompleted = !IsInitialBlockDownload();
```

Comment:
Can you move this update to UpdatedBlockTip? (it already has an fInitialDownload instead of having to call IsInitialDownload, even).

Reply:
This code part doesn't add a new call, it just preserves the state of the result of `IsInitialBlockDownload()` in the new `IsInitialBlockDownload` variable.

Reply:
Yes, I'm suggesting you move the update of initialBlockDownloadCompleted to UpdatedBlockTip instead of BlockChecked.